bofire/strategies/doe/design.py

```diff
@@ -0,0 +1,255 @@
+    n_experiments: Optional[int] = None,
+    tol: float = 0,
+    delta: float = 1e-7,
+    ipopt_options: Dict = {},
+    jacobian_building_block: Optional[Callable] = None,
+    sampling: Optional[np.ndarray] = None,
+    fixed_experiments: Optional[np.ndarray] = None,
+) -> pd.DataFrame:
+    """Function computing a d-optimal design" for a given opti problem and model.
+
+    Args:
+        problem (opti.Problem): problem containing the inputs and constraints.
+        model_type (str, Formula): keyword or formulaic Formula describing the model. Known keywords
+            are "linear", "linear-and-interactions", "linear-and-quadratic", "fully-quadratic".
+        n_experiments (int): Number of experiments. By default the value corresponds to
+            the number of model terms - dimension of ker() + 3.
+        tol (float): Tolerance for linear/NChooseK constraint violation. Default value is 0.
+        delta (float): Regularization parameter. Default value is 1e-3.
+        ipopt_options (Dict): options for IPOPT. For more information see [this link](https://coin-or.github.io/Ipopt/OPTIONS.html)
+        jacobian_building_block (Callable): Only needed for models of higher order than 3. derivatives
+            of each model term with respect to each input variable.
+        sampling (Sampling, np.ndarray): Sampling class or a np.ndarray object containing the initial guess.
+        fixed_experiments (np.ndarray): numpy array containing experiments that will definitely part of the design.
+            Values are set before the optimization.
+
+    Returns:
+        A pd.DataFrame object containing the best found input for the experiments. This is only a
+        local optimum.
+
+    """
+
+    try:
+        from cyipopt import minimize_ipopt  # noqa
+    except ImportError as e:
+        warnings.warn(e.msg)
+        warnings.warn(
+            "please run `conda install -c conda-forge cyipopt` for this functionality."
+        )
+        raise e
+
+    assert all(
+        [c.min_count == 0 for c in domain.cnstrs if isinstance(c, NChooseKConstraint)]
+    ), "NChooseKConstraint with min_count !=0 is not supported!"
+
+    # determine number of experiments
+    n_experiments_min = (
+        len(
+            get_formula_from_string(
+                model_type=model_type, rhs_only=True, domain=domain
+            ).terms
+        )
+        + 3
+    )
+
+    D = len(domain.inputs)
+    model_formula = get_formula_from_string(
+        model_type=model_type, rhs_only=True, domain=domain
+    )
+
+    if n_experiments is None:
+        n_experiments = n_experiments_min
+    elif n_experiments < n_experiments_min:
+        warnings.warn(
+            f"The minimum number of experiments is {n_experiments_min}, but the current setting is n_experiments={n_experiments}."
+        )
+
+    # initital values
+    if sampling is not None:
+        x0 = sampling
+    else:
+        domain_for_sampling = _domain_for_sampling(domain=domain)
```

Comment:
As described above: I think this is not anymore necessary if you merge the current main in, and use the Polytopesampler as is.